### PR TITLE
Create TF Module that implements public/private `subnet` strategy in existing VPC

### PR DESCRIPTION
**Pull request by s2504s**
## What
* Create TF Module that implements a public & private`subnet` strategy using existing VPC

## Why
* Encapsulate business logic of VPCs and Subnets in a module

### CODE REVIEW

**Comment by osterman** [comment]
@s2504s looks great! Just add `README.md`

**Comment by comeanother** [comment]
There was an issue if we try to use existing AWS Internet Gateway and Default Route Table.

Even if we use code like this it doesn't work.
```
resource "aws_route_table_association" "public" {
  count = "${length(var.availability_zones)}"
  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
  route_table_id = "${signum(length(var.vpc_default_route_table)) == 1 ? var.vpc_default_route_table : aws_route_table.public.id }"
}
```

Because if we do not create `aws_route_table.public` (when we use existing route table), Terraform is unable to resolve `aws_route_table.public.id` and we got an error.

```
Error running plan: 1 error(s) occurred:

* module.tf_subnets.aws_route_table_association.public: 2 error(s) occurred:

* module.tf_subnets.aws_route_table_association.public[0]: Resource 'aws_route_table.public' not found for variable 'aws_route_table.public.id'
* module.tf_subnets.aws_route_table_association.public[1]: Resource 'aws_route_table.public' not found for variable 'aws_route_table.public.id'
```

I've pinned `igw_id` as required value and added resources for two cases: use existing `aws_route_table` and create new `aws_route_table`.
```
resource "aws_route_table_association" "public_new" {
  count = "${signum(length(var.vpc_default_route_table)) == 1 ? 0 : length(var.availability_zones)}"
  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
  route_table_id = "${aws_route_table.public.id}"
}

resource "aws_route_table_association" "public_exists" {
  count = "${signum(length(var.vpc_default_route_table)) == 1 ? length(var.availability_zones) : 0}"
  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
  route_table_id = "${var.vpc_default_route_table}"
}
```

**Comment by comeanother** [comment]
I've tested `terraform plan` with two cases:
```
module "tf_subnets" {
  source = "git::https://github.com/cloudposse/tf_subnets.git?ref=init"

  availability_zones      = ["us-east-1a", "us-east-1b"]
  namespace               = "test"
  name                    = "test"
  stage                   = "test"
  region                  = "us-east-1"
  vpc_id                  = "vpc-aceb27ca"
  igw_id                  = "igw-9c26adfb"
  vpc_default_route_table = "rtb-f4f0ce92"
}
```

```
module "tf_subnets" {
  source = "git::https://github.com/cloudposse/tf_subnets.git?ref=init"

  availability_zones      = ["us-east-1a", "us-east-1b"]
  namespace               = "test"
  name                    = "test"
  stage                   = "test"
  region                  = "us-east-1"
  vpc_id                  = "vpc-aceb27ca"
  igw_id                  = "igw-9c26adfb"
}
```

Both are ok.

**Comment by comeanother** [comment]
@osterman please review

File: public.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+resource "aws_route_table_association" "public_new" {
+  count = "${signum(length(var.vpc_default_route_table)) == 1 ? 0 : length(var.availability_zones)}"
+  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
+  route_table_id = "${aws_route_table.public.id}"
+}
+
+resource "aws_route_table_association" "public_exists" {
+  count = "${signum(length(var.vpc_default_route_table)) == 1 ? length(var.availability_zones) : 0}"
+  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
+  route_table_id = "${var.vpc_default_route_table}"
```
Comment (osterman) [inline public.tf]: We should suffix with `_id` if it refers to an ID (for consistency).

File: public.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  tags = "${module.tf_label.tags}"
+}
+
+resource "aws_route_table_association" "public_new" {
+  count = "${signum(length(var.vpc_default_route_table)) == 1 ? 0 : length(var.availability_zones)}"
+  subnet_id      = "${element(aws_subnet.public.*.id, count.index)}"
+  route_table_id = "${aws_route_table.public.id}"
+}
+
+resource "aws_route_table_association" "public_exists" {
```
Comment (osterman) [inline public.tf]: I think we should call this `public_default`

File: public.tf [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  vpc_id = "${data.aws_vpc.default.id}"
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = "${var.igw_id}"
+  }
+  tags = "${module.tf_label.tags}"
+}
+
+resource "aws_route_table_association" "public_new" {
```
Comment (osterman) [inline public.tf]: Let's call this `public`  so it's symetrical with `aws_route_table.public`

File: public.tf (hunk 1 above)
Comment (comeanother) [inline public.tf]: Added.

File: public.tf (hunk 2 above)
Comment (comeanother) [inline public.tf]: Renamed.

File: public.tf (hunk 3 above)
Comment (comeanother) [inline public.tf]: Renamed.

**Comment by comeanother** [comment]
Added required fixes.
@osterman please approve.

**Pull request merged**